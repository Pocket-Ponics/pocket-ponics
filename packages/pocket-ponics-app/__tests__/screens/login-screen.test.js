import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Platform, TextInput } from 'react-native'
import AuthUtil from '../../util/auth-util'
jest.mock('../../util/auth-util')

import LoginScreen from '../../screens/login-screen'

describe('Login Screen', () => {
	test('renders correctly', () => {
		Platform.OS = ''
		const tree = shallow(
			<LoginScreen />
		).html()
		expect(tree).toMatchSnapshot()
	})

	test('navigates to other screens', () => {
		Platform.OS = 'ios'
		const navigation = {
			navigate: jest.fn()
		}
		const tree = shallow(
			<LoginScreen navigation={navigation}/>
		)

		tree.find(TouchableOpacity).at(0).props().onPress()
		// TouchableOpacity @ 1 logs user in
		tree.find(TouchableOpacity).at(2).props().onPress()
		tree.find(TouchableOpacity).at(3).props().onPress()
		expect(navigation.navigate).toHaveBeenCalledTimes(3)
	})

	test('tries to log in with no password', () => {
		const tree = shallow(
			<LoginScreen />
		)

		tree.find(TextInput).at(0).props().onChangeText('mockUser')
		tree.find(TextInput).at(1).props().onChangeText('')
		tree.find(TouchableOpacity).at(1).props().onPress()
		expect(tree.html()).toMatchSnapshot()
	})

	test('logs in', () => {
		const tree = shallow(
			<LoginScreen  navigation={{ navigate: jest.fn() }}/>
		)
		AuthUtil.login.mockImplementation((user, pass, fn) => fn())

		tree.find(TextInput).at(0).props().onChangeText('mockUser')
		tree.find(TextInput).at(1).props().onChangeText('mockPassword')
		tree.find(TouchableOpacity).at(1).props().onPress()
		expect(AuthUtil.login).toHaveBeenCalled()
	})
})