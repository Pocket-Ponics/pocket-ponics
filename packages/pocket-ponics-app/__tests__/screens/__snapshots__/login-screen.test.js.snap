// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders correctly 1`] = `"<View style=\\"flex:1;background-color:#472600\\"><View style=\\"flex:1;align-items:center;justify-content:center;padding:20px;margin-top:40px;width:100%\\"><View style=\\"align-self:flex-end;padding:12px;opacity:1\\"><Text></Text></View><Image source=\\"1\\" style=\\"height:200px;width:200px;resize-mode:contain\\"></Image><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Email Address\\" placeholderTextColor=\\"#FFFFFF\\" value=\\"\\" autoCapitalize=\\"none\\" textContentType=\\"emailAddress\\"></TextInput></View><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Password\\" placeholderTextColor=\\"#FFFFFF\\" value=\\"\\" autoCapitalize=\\"none\\" textContentType=\\"password\\"></TextInput></View><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;width:100%;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Log In</Text></View><View style=\\"opacity:1\\"><Text style=\\"color:#84AB71;font-size:18px;margin-bottom:10px\\">Sign Up</Text></View><View style=\\"opacity:1\\"><Text style=\\"color:#84AB71;font-size:18px;margin-bottom:10px\\">Reset Password</Text></View><ActivityIndicator color=\\"#FFFFFF\\"></ActivityIndicator></View></View>"`;

exports[`Login Screen tries to log in with no password 1`] = `"<View style=\\"0:[object Object];1:[object Object]\\"><View style=\\"flex:1;align-items:center;justify-content:center;padding:20px;margin-top:40px;width:100%\\"><View style=\\"align-self:flex-end;padding:12px;opacity:1\\"><Text></Text></View><Image source=\\"1\\" style=\\"height:200px;width:200px;resize-mode:contain\\"></Image><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Email Address\\" placeholderTextColor=\\"#FFFFFF\\" value=\\"mockUser\\" autoCapitalize=\\"none\\" textContentType=\\"emailAddress\\"></TextInput></View><View style=\\"margin-top:5px;align-items:center;width:100%\\"><Text></Text><TextInput underlineColorAndroid=\\"transparent\\" style=\\"width:100%;height:45px;border-radius:45px;font-size:16px;padding-left:45px;background-color:rgba(0, 0, 0, 0.35);color:#FFFFFF\\" placeholder=\\"Password\\" placeholderTextColor=\\"#FFFFFF\\" value=\\"\\" autoCapitalize=\\"none\\" textContentType=\\"password\\"></TextInput></View><View style=\\"background-color:#456E31;border-radius:5px;padding:10px;margin:10px;width:100%;align-items:center;opacity:1\\"><Text style=\\"color:#FFFFFF;font-size:22px\\">Log In</Text></View><View style=\\"opacity:1\\"><Text style=\\"color:#84AB71;font-size:18px;margin-bottom:10px\\">Sign Up</Text></View><View style=\\"opacity:1\\"><Text style=\\"color:#84AB71;font-size:18px;margin-bottom:10px\\">Reset Password</Text></View><ActivityIndicator color=\\"#FFFFFF\\"></ActivityIndicator></View></View>"`;
